db/migrate/20180705230311_smart_proxy_capabilities.rb

```diff
@@ -0,0 +1,17 @@
+class SmartProxyCapabilities < ActiveRecord::Migration[5.1]
+  def up
+    rename_table :features_smart_proxies, :smart_proxy_features
+    change_table :smart_proxy_features, :bulk => true do |t|
+      t.primary_key :id
+      t.text :capabilities, array: true
+      t.text :settings
+    end
+  end
+
+  def down
+    change_table :smart_proxy_features do |t|
+      t.remove :settings, :capabilities, :id
+    end
+    rename_table :smart_proxy_features, :features_smart_proxies
+  end
+end
```

Comment:
The `down` is not on par with `up`, I am missing removal of capabilities text-array column here.

Reply:
I'm not sure what you mean, its right above there?  Maybe you are looking at an older commit?

Reply:
Oh that was it yeah.

Reply:
What is the upgrade plan? I think we can simply drop all features and refresh them from proxies, the question is when to do that. We do not require smart proxies to be available during upgrade. Another option is to migrate the data - convert associations to primary key.

Reply:
Any user using the installer will have their smart proxies refreshed automatically when they upgrade either the main foreman server, or the specific smart proxies.  For users not using the installer, i think a manual refresh is fine.  We can document this.  Keep in mind that as of now everything will continue to work fine even without a refresh since this doesn't remove support for running a proxy with just the v1 api.

Reply:
AFAIK currently we don't have explicit instructions on the upgrade order. Some users upgrade foreman -> proxy and others proxy -> foreman. Both should work but in the latter case we don't have capabilities. Adding an upgrade note about this is IMHO sufficient because we don't rely on capabilities (yet).